pkg/storage/fs/ocis/path.go

```diff
@@ -90,12 +94,21 @@ func (pw *Path) NodeFromID(ctx context.Context, id *provider.ResourceId) (n *Nod
 	if id == nil || id.OpaqueId == "" {
 		return nil, fmt.Errorf("invalid resource id %+v", id)
 	}
-	return ReadNode(pw, id.OpaqueId)
+	return ReadNode(ctx, pw, id.OpaqueId)
 }
 
 // Path returns the path for node
 func (pw *Path) Path(ctx context.Context, n *Node) (p string, err error) {
-	for !n.IsRoot() {
+	var root *Node
+	if pw.EnableHome {
+		root, err = pw.HomeNode(ctx)
```

Comment:
third deja vu. move this condition into utility method ?

Reply:
decided on `pw.HomeOrRootNode(ctx)`